lib/engine/game/g_18_india/step/assign.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/assign'
+
+module Engine
+  module Game
+    module G18India
+      module Step
+        class Assign < Engine::Step::Assign
+          # Allow company 'P6' to assign to any hex that isn't a town or city
+          def available_hex(entity, hex)
+            return unless entity.company?
+            return unless entity.id == 'P6'
+            return if hex.assigned?(entity.id)
+            return unless hex.tile.city_towns.empty? # hex doesn't have a town or city
+
+            @game.hex_by_id(hex.id).neighbors.keys
```

Comment:
This method should just be returning a boolean value. Can you just do this?
```suggestion
            hex.tile.city_towns.empty?
```

Reply:
I modified the base code from Engine::Step::Assign. 
I added to the guard clauses. 

https://github.com/tobymao/18xx/blob/02f8ebe84a373e5768337512c2de0a45f1fede76/lib/engine/step/assign.rb#L67

Reply:
Tested the modification.  It seems to work.  I will wonder if the base code I inherited this from should be updated.

Reply:
The base code might be written this way to check that there are some neighbouring hexes, to make sure you can't select a dead hex that can never be connected to anything. Could this be a problem here?

Reply:
Your suggested code work when tested.  The only hex I saw that needed to be excluded was the blue water hex, added a guard clause.